docs/api/js/types.md

```diff
@@ -1,226 +1,303 @@
+
+- `create`
+- `createAccount`
+
+## `Eip712Meta`
+
+Type defining an EIP-712 transaction with optional parameters.
+
+- `gasPerPubdata?`: `BigNumberish`;
+- `factoryDeps?`: `BytesLike[]`;
+- `customSignature?`: `BytesLike`;
+- `paymasterParams?`: `PaymasterParams`;
+
+## `EthereumSignature`
+
+Interface representation of an Ethereum signature.
+
+- `v`: `number`;
+- `r`: `BytesLike`;
+- `s`: `BytesLike`;
+
+## `EventFilter`
+
+Interface representation of event filter containing various fields.
+
+- `topics?`: `Array<string | Array<string> | null>`;
+- `address?`: `Address | Array<Address>`;
+- `fromBlock?`: `BlockTag`;
+- `toBlock?`: `BlockTag`;
+- `blockHash?`: `string`;
+
+## `FullDepositFee`
+
+Interface representation of full deposit fee containing various mandatory and optional fields.
+
+- `maxFeePerGas?`: `BigNumber`;
+- `maxPriorityFeePerGas?`: `BigNumber`;
+- `gasPrice?`: `BigNumber`;
+- `baseCost`: `BigNumber`;
+- `l1GasLimit`: `BigNumber`;
+- `l2GasLimit`: `BigNumber`;
+
+## `GeneralPaymasterInput`
+
+Interface representation of general paymaster input containing a couple of fields, including a `General` type.
+
+- `type`: `'General'`;
+- `innerInput`: `BytesLike`;
+
+## `L2ToL1Log`
+
+Interface representation of a layer 2 to layer 1 transaction log containing various fields.
+
+- `blockNumber`: `number`;
+- `blockHash`: `string`;
+- `l1BatchNumber`: `number`;
+- `transactionIndex`: `number`;
+- `shardId`: `number`;
+- `isService`: `boolean`;
+- `sender`: `string`;
+- `key`: `string`;
+- `value`: `string`;
+- `transactionHash`: `string`;
+- `logIndex`: `number`;
+
+## `Log`
+
+Interface representation of log that extends Ethers `providers.Log` and supplies the layer 1 batch number.
+
+- `l1BatchNumber`: `number`;
+
+## `MessageProof`
+
+Interface representation of message proof containing various fields.
+
+- `id`: `number`;
+- `proof`: `string[]`;
+- `root`: `string`;
+
+## `Network`
+
+Enumerated list of networks and their ids.
+
+- Mainnet = `1`
+- Ropsten = `3`
+- Rinkeby = `4`
+- Goerli = `5`
+- localhost = `9`
+
+## `PaymasterInput`
+
+Type definition for a paymaster input specified as either approval based or general.
+
+- `ApprovalBasedPaymasterInput` | `GeneralPaymasterInput`
+
+## `PaymasterParams`
+
+Type defining a paymaster by address and the bytestream input.
+
+- `paymaster`: `Address`;
+- `paymasterInput`: `BytesLike`;
+
+## `PriorityOpResponse`
+
+Interface representation of priority op response that extends [`TransactionResponse`](#transactionresponse) and adds a function that waits to commit a layer 1 transaction, including when given on optional confirmation number.
+
+- `waitL1Commit(confirmation?: number)`: `Promise<providers.TransactionReceipt>`;
+
+## `PriorityOpTree`
+
+Enumerated list of priority op tree types.
+
+- Full = `0`
+- Rollup = `1`
+
+## `PriorityQueueType`
+
+Enumerated list of priority queue types.
+
+- Deque = `0`
+- HeapBuffer = `1`
+- Heap = `2`
```

Comment:
This is legacy that should be removed from both SDK and docs. I will prepare the PR for SDK, in the meanwhile please remove it

Reply:
Done